### DISCUSSION TITLE
Websockets Debugging

### DISCUSSION THREAD

**Discussion by bmschwa** [photoprism/photoprism#4358] [Q&A]
Hello all,

I was hoping the community could offer some ideas about how I could debug a problem I'm having with web sockets.

Specifically, I'm using a [synology nas](https://docs.photoprism.app/getting-started/nas/synology/) to run a docker container and then I have the [synology web station](https://kb.synology.com/en-us/DSM/help/WebStation/application_webserv_desc?version=7) enabled in order to secure the site.

I was convinced I had misconfigured a component in synology/web station after reading about [reverse proxies](https://docs.photoprism.app/getting-started/proxies/traefik/#why-use-a-proxy).  *However* when I have these settings set I'm seeing activity on the server side indicating the requests are getting through.

```
PHOTOPRISM_DEBUG='true'
PHOTOPRISM_LOG_LEVEL='debug'
PHOTOPRISM_SITE_URL='https://the-right-domain.net'
PHOTOPRISM_DEFAULT_TLS='false'
PHOTOPRISM_DISABLE_TLS='true'
````

In the photoprism output, I'm seeing requests to the right endpoint....
```
DEBU[2024-06-27T21:00:09Z] server: GET /api/v1/ws (400) [34.12µs]
DEBU[2024-06-27T21:00:15Z] server: GET /api/v1/ws (400) [30.15µs]
DEBU[2024-06-27T21:00:21Z] server: GET /api/v1/ws (400) [31.501µs]
DEBU[2024-06-27T21:00:27Z] server: GET /api/v1/ws (400) [33.09µs]
```

And on the client side, the 400/failures are visible in the console.

My best guess is that something is there is an encryption issue (traffic being sent through without being decrypted) by web station.... but I really don't know.   Anyone have ideas about how to confirm the hypothesis or otherwise debug?

**Comment by bmschwa** [comment]
Hey @mikelagase, Saw you were running photoprism on a synology device ([this discussion](https://github.com/photoprism/photoprism/discussions/4363)).   Were you able to get the reverse proxy configured?

**Comment by bmschwa** [comment]
@lastzero, I've seen you solve a couple websockets inquiries ([1535](https://github.com/photoprism/photoprism/discussions/1535) & [1607](https://github.com/photoprism/photoprism/discussions/1607) for instance).... I'm seeing the websocket request making it to the container, but I don't know how to debug from there.   Can you point me in the right direction?

Thanks,

**Comment by alexisLefebvre** [comment]
Did you enable the “Upgrade” options? It worked while using a different software (Outline) but that may be the key to allow websockets, see https://docs.getoutline.com/s/hosting/doc/nginx-6htaRboR57
